src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1663,12 +1808,40 @@ def __deepcopy__(self, memodict={}) -> "SetStatement":  # noqa: B006
     @property
     def sql_query(self) -> str:
         if not self._sql_query:
-            sql = f"({self.set_operands[0].selectable.sql_query})"
-            for i in range(1, len(self.set_operands)):
-                sql = f"{sql}{self.set_operands[i].operator}({self.set_operands[i].selectable.sql_query})"
-            self._sql_query = sql
+            sql_query = (
+                self._generate_sql_with_comments()
+                if self._session._generate_multiline_queries
+                else self._generate_sql_without_comments()
```

Comment:
great that we are param protecting this. Let's make this param False by default when reading from session